### PR TITLE
Validate spec models

### PR DESCRIPTION
**Pull request by AmineDerbal**
In this pull request ♻️ 

- Add required validations for both user and post models 👍 
- Add custom validation for the other models 👍 
- Create unit spec for all models methods and validations using RSPEC 👍 
- Made sure all the tests are working 👍

### CODE REVIEW

File: spec/comment_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  describe '#update_post_comments_counter' do
+    let(:user) { User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.', posts_counter: 0) }
+    let!(:post) { user.posts.create(title: 'Post 1', text: 'This is post 1', comments_counter: 0, likes_counters: 0) }
+    let!(:comment) { post.comments.create(text: 'Comment 1', author_id: user.id) }
+
+    it 'user_counter' do
+      user.increment!(:posts_counter)
+      expect(user.posts_counter).to eq(1)
+    end
```
Comment (addod19) [inline spec/comment_spec.rb]: - [x] I have not seen this method tested `update_post_comments_counter`, kindly test the method the way you tested the  way you tested the `recent_comments` in the Post model

File: spec/like_spec.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  describe '#update_post_likes_counter' do
+    let(:user) { User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.', posts_counter: 0) }
+    let!(:post) { user.posts.create(title: 'Post 1', text: 'This is post 1', comments_counter: 0, likes_counters: 0) }
+    let!(:like) { post.likes.create(author_id: user.id) }
+
+    it 'user_counter' do
+      user.increment!(:posts_counter)
+      expect(user.posts_counter).to eq(1)
+    end
```
Comment (addod19) [inline spec/like_spec.rb]: - [x] Here also, I have not seen the method `update_post_likes_counter` tested in the spec file. The emphasis is on the `method`

**Review by addod19 — CHANGES_REQUESTED** [review]
### Hi @AmineDerbal, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [ ] Correct Git-flow followed :100: 
- [ ] README file is professional and has all relevant headers :1st_place_medal: 
- [ ] No linter errors recorded :+1: 
- [ ] Validations tests implemented :rocket: 
- [ ] Model validations implemented as expected :fireworks: 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Going forward, let us kindly capitalize all our commit messages :man_technologist: 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/comment_spec.rb (hunk 1 above)
Comment (AmineDerbal) [reply spec/comment_spec.rb]: Got it

File: spec/like_spec.rb (hunk 2 above)
Comment (AmineDerbal) [reply spec/like_spec.rb]: done

**Comment by AmineDerbal** [comment]
- Update the spec units to test the methods of increment correctly 👍

File: spec/comment_spec.rb (hunk 1 above)
Comment (rloterh) [reply spec/comment_spec.rb]: well resolved! ✔️

File: spec/like_spec.rb (hunk 2 above)
Comment (rloterh) [reply spec/like_spec.rb]: well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Amine,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Good job implementing recommended changes!

Cheers..Happy coding!💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**